### PR TITLE
add documentation for unit testing incremental models

### LINKED ISSUE #4929: Docs for unit testing incremental models
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

New section on this page 
https://docs.getdbt.com/docs/build/unit-tests

### What part(s) of the page would you like to see updated?

New section on unit testing incremental models

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by graciegoheen**
## What are you changing in this pull request and why?

Closes #4929

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/static/_redirects`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

File: website/docs/reference/resource-properties/unit-tests.md [hunk 1]
```diff
@@ -153,6 +153,10 @@ When `format: csv`, can either supply:
     - If you do supply an input for a seed, we will use that input instead.
 - You can also have “empty” inputs, by setting rows to an empty list `rows: []`
 
+### Overrides
+
+When configuring your unit test, you can override the output of macros, vars, or environment variables for a given unit test.
```
Comment (graciegoheen) [inline website/docs/reference/resource-properties/unit-tests.md:158]: Should probably have more examples / info here - but added this as a stub for now

added to follow up issue https://github.com/dbt-labs/docs.getdbt.com/issues/4889

**Review by MichelleArk — APPROVED** [review]
Content looks spot-on!

File: website/docs/docs/build/unit-tests.md [hunk 2]
```diff
@@ -254,5 +254,72 @@ unit_tests:
 
 ```
 
+## Unit testing incremental models
 
+When configuring your unit test, you can override the output of macros, vars, or environment variables for a given unit test. This enables you to unit test your incremental models - both in "full refresh" and "incremental" mode.
```
Comment (matthewshaver) [inline website/docs/docs/build/unit-tests.md]:
```suggestion
When configuring your unit test, you can override the output of macros, vars, or environment variables. This enables you to unit test your incremental models in "full refresh" and "incremental" modes. 
```

File: website/docs/docs/build/unit-tests.md [hunk 3]
```diff
@@ -254,5 +254,72 @@ unit_tests:
 
 ```
 
+## Unit testing incremental models
 
+When configuring your unit test, you can override the output of macros, vars, or environment variables for a given unit test. This enables you to unit test your incremental models - both in "full refresh" and "incremental" mode. 
 
+When testing an incremental model, the expected output is the __result of the materialization__ (i.e. what will be merged/inserted), not the resulting model itself (i.e. what the final table will look like after the merge/insert).
```
Comment (matthewshaver) [inline website/docs/docs/build/unit-tests.md]:
```suggestion
When testing an incremental model, the expected output is the __result of the materialization__ (what will be merged/inserted), not the resulting model itself (what the final table will look like after the merge/insert).
```

File: website/docs/docs/build/unit-tests.md [hunk 4]
```diff
@@ -254,5 +254,72 @@ unit_tests:
...
+
+Let's say I have an incremental model in my project:
```
Comment (matthewshaver) [inline website/docs/docs/build/unit-tests.md]:
```suggestion
For example, say you have an incremental model in your project:
```

File: website/docs/docs/build/unit-tests.md [hunk 5]
```diff
@@ -254,5 +254,72 @@ unit_tests:
...
+}}
+
+select * from {{ ref('events') }}
+{% if is_incremental() %}
+where event_time > (select max(event_time) from {{ this }})
+{% endif %}
+
+```
+
+I can define unit tests on `my_incremental_model` to ensure my incremental logic is working as expected:
```
Comment (matthewshaver) [inline website/docs/docs/build/unit-tests.md]:
```suggestion
You can define unit tests on `my_incremental_model` to ensure your incremental logic is working as expected:
```

File: website/docs/docs/build/unit-tests.md [hunk 6]
```diff
@@ -254,5 +254,72 @@ unit_tests:
...
+          - {event_id: 1, event_time: 2020-01-01}
+    expect:
+      # what will be inserted/merged into my_incremental_model
+      rows:
+        - {event_id: 2, event_time: 2020-01-02}
+        - {event_id: 3, event_time: 2020-01-03}
+
+```
+
+There is currently no way to unit test whether the dbt framework actually inserted/merged the records into your existing model correctly or not - but [we're investigating a way to support this in the future](https://github.com/dbt-labs/dbt-core/issues/8664).
```
Comment (matthewshaver) [inline website/docs/docs/build/unit-tests.md]:
```suggestion
There is currently no way to unit test whether the dbt framework inserted/merged the records into your existing model correctly, but [we're investigating support for this in the future](https://github.com/dbt-labs/dbt-core/issues/8664).
```

**Review by matthewshaver — APPROVED** [review]

**Comment by graciegoheen** [comment]
@matthewshaver any idea why the vercel deployment is failing?

**Comment by matthewshaver** [comment]
> @matthewshaver any idea why the vercel deployment is failing?

It's redeploying now.

File: website/docs/docs/build/unit-tests.md [hunk 7]
```diff
@@ -254,5 +254,74 @@ unit_tests:
...
 
+## Unit testing incremental models
 
+When configuring your unit test, you can override the output of macros, vars, or environment variables. This enables you to unit test your incremental models in "full refresh" and "incremental" modes. 
 
+When testing an incremental model, the expected output is the __result of the materialization__ (what will be merged/inserted), not the resulting model itself (what the final table will look like after the merge/insert).
+
+For example, say you have an incremental model in your project:
+
+<file name='my_incremental_model.sql'>
```
Comment (matthewshaver) [inline website/docs/docs/build/unit-tests.md]:
```suggestion
<File name='my_incremental_model.sql'>
```

File: website/docs/docs/build/unit-tests.md [hunk 8]
```diff
@@ -254,5 +254,74 @@ unit_tests:
...
+}}
+
+select * from {{ ref('events') }}
+{% if is_incremental() %}
+where event_time > (select max(event_time) from {{ this }})
+{% endif %}
+
+```
+
+</file>
```
Comment (matthewshaver) [inline website/docs/docs/build/unit-tests.md]:
```suggestion
</File>
```

**Pull request merged**